src/retsnoop.c

```diff
@@ -196,9 +204,55 @@ static int append_str_file(char ***strs, int *cnt, const char *file)
 	return err;
 }
 
-static int append_compile_unit(char ***strs, int *cnt, const char *compile_unit)
+static int append_compile_unit(struct ctx *ctx, char ***strs, int *cnt, const char *compile_unit)
 {
 	int err = 0;
+
```

Comment:
nit: C89 style dictates that all variable declarations should be at the top of the function, no empty lines between them

Reply:
Ouch!  I haven't worked on C89 style for long.

Reply:
yeah, we need to add --std=gnu89 to automatically enforce that, just like I just did for libbpf: https://github.com/libbpf/libbpf/pull/408